pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -88,9 +88,11 @@ func ValidateKubeSchedulerConfiguration(cc *config.KubeSchedulerConfiguration) u
 			}
 		}
 	}
-	if cc.PercentageOfNodesToScore < 0 || cc.PercentageOfNodesToScore > 100 {
-		errs = append(errs, field.Invalid(field.NewPath("percentageOfNodesToScore"),
-			cc.PercentageOfNodesToScore, "not in valid range [0-100]"))
+	if cc.PercentageOfNodesToScore != nil {
+		if *cc.PercentageOfNodesToScore < 0 || *cc.PercentageOfNodesToScore > 100 {
+			errs = append(errs, field.Invalid(field.NewPath("percentageOfNodesToScore"),
+				cc.PercentageOfNodesToScore, "not in valid range [0-100]"))
+		}
```

Comment:
We can move the code out to a function so as to reuse here and L176~L181:

```go
func validatePercentageOfNodesToScore(path *field.Path, percentageOfNodesToScore *int32) error {
	if percentageOfNodesToScore != nil {
		if *percentageOfNodesToScore < 0 || *percentageOfNodesToScore > 100 {
			return field.Invalid(path, percentageOfNodesToScore, "not in valid range [0-100]")
		}
	}
	return nil
}

# then here we call:
errs = append(errs, validatePercentageOfNodesToScore(field.NewPath("percentageOfNodesToScore"), cc.PercentageOfNodesToScore))

# in validateKubeSchedulerProfile() we call:
errs = append(errs, validatePercentageOfNodesToScore(field.NewPath("percentageOfNodesToScore"), profile.PercentageOfNodesToScore))
```

Reply:
Made the suggested change.